pages/index.js

```diff
@@ -7,7 +8,23 @@ import calloutImg from "../public/images/callOutBox.jpeg";
 import CategoriesSelection from "~/components/CategorySelection";
 import Categories from "./categories";
 import HeaderConditionSection from "~/components/HeaderConditionSection";
+import { useState } from "react";
+
 const Index = () => {
+  const { data: issuesCategories } = useSWR("/api/issues");
+  const [searchInput, setSearchInput] = useState("");
+
+  let filteredIssuesData =
+    issuesCategories &&
+    issuesCategories.data.filter((entry) => {
+      return (
+        entry.name.toLowerCase().includes(searchInput.toLowerCase()) ||
+        entry.description.toLowerCase().includes(searchInput.toLowerCase()) ||
+        entry.slug.toLowerCase().includes(searchInput.toLowerCase())
```

Comment:
Let's try and neaten this up a little. You could:

- destructure entry on line 19
- then put what you want to check in an array, ie `[name, description, slug]`
- then use `Array.some()` on that array to check each one for what you're interested in

This would mean line 21 would look something like:
`return [name, description...].some(string => this is where you compare the string with what you're searching for)`

Reply:
Hi @ARTurnerGit, Thank you for the feedback,  I still don't understand how should the code look like and why we are doing this.

Reply:
I tried this and it didn't work:
```
  let filteredIssuesData =
    issuesCategories &&
    issuesCategories.data.filter((entry) => {
      return (
       (entry) => [name, description, slug].some(string =>string.toLowerCase().includes(searchInput.toLowerCase()))
      );
    });
```

Reply:
What I mean was the first argument to the callback can be destructured. So it would look something like this (I may have made some bracket errors

`filter(({name, description, slug})=>[name, description, slug].some(string => string.toLowerCase().includes(searchInput.toLowerCase())))`

Reply:
It's just going to be the equivalent to what you have - if you can't see how it's working don't worry about it, keep the code that you had before. It's not worth putting in code you aren't 100% sure of just to make it neater.

Reply:
Array.some will return true if any of the entries in the array (in this case name, description, slug or whatever else you put in there) pass the condition in the callback (ie checking if one string is inside another string)

Reply:
Make sense, thank you. I now understand it. I'll change the code